src/commands/exec.ts

```diff
@@ -240,24 +245,36 @@ export default class Exec extends BaseCommand {
   async run(): Promise<void> {
     inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
 
-    const { flags } = await this.parse(Exec);
+    const { args, flags } = await this.parse(Exec);
+
+    // Automatically set tty if the user doesn't supply it based on whether stdin is TTY.
+    if (flags.tty === undefined) {
+      if (process.stdin.isTTY) {
+        flags.tty = true;
+      } else {
+        flags.tty = false;
+        this.log(chalk.yellow('stdin does not support tty and was automatically disabled, --no-tty should be used'));
```

Comment:
Hmm I know I said this should be a warning, but if we detect the stdin doesn't support tty then do they really need to specify `--no-tty`?